cli/aiconfig-editor/src/shared/types.ts

```diff
@@ -1,7 +1,43 @@
+import { AIConfig, Prompt, Error, ExecuteResult } from "aiconfig";
+
 export type EditorFile = {
   name: string;
   extension: string;
   path: string;
   isDirectory: boolean;
   disabled?: boolean;
 };
+
+export type TextExecuteResultOutput = {
+  type: "text";
+  text: string;
+};
+
+export type ClientExecuteResultOutput = TextExecuteResultOutput; // TODO: Add non-text types
```

Comment:
Not blocking for this diff, but this is a bit tricky. I'm wondering if we can avoid this type altogether by having a standardized format with which to store outputs. 

The other thing I'm concerned about at the moment is handling things like function calling and other more complex flows. We don't have to support everything now, but just design it such that they could be supported.

Ideally we could just use the main AIConfig type instead of needing this kind of intermediate type.

Reply:
Ya, left a comment above. @rossdanlm is looking into implementing a generalized Output type that will allow us to get the text / mime-type data directly, and once that's done we can come back and ideally remove all of this.

Note that this is only being done to support outputs right now -- function calling properties are already supported in model settings. I'm not sure what the expectation will be for those within the editor itself though since it needs a python runtime (i.e. running code outside of AIConfig) to call the functions, right?

Reply:
Yeah, mostly just thinking about how it gets rendered for an aiconfig that contains `functions`. The output in that case would contain the function call to execute, so just thinking about how it will get rendered by the editor

Reply:
Ah, I see. Would it make sense to also treat function calls as just text to show the literal function call? They would probably be used to then propagate the results into the config via parameters in most cases, right? It's tough to say what the best UI would be for that output otherwise